src/main/java/com/gooddata/executeafm/afm/PopMeasureDefinition.java

```diff
@@ -68,9 +87,58 @@ public ObjQualifier getPopAttribute() {
         return popAttribute;
     }
 
+    /**
+     * Returns the qualifier used by the {@link com.gooddata.md.Metric}.
+     *
+     * @return qualifier used by the metric, which is its {@link #popAttribute}.
+     *
+     * @throws IllegalStateException
+     *         The exception is thrown when {@link #popAttribute} has not been set.
+     * @deprecated Use {@link #getObjQualifiers()} instead.
+     */
     @Override
+    @Deprecated
     public ObjQualifier getObjQualifier() {
-        return getPopAttribute();
+        return getObjQualifiers().stream()
```

Comment:
why do you change this method? Apparently it was sufficient to just return null till now. Moreover couldn't it be implemented in default method instead?

Reply:
Reverted to the original method. I wanted to fall back on the new method. Possibly, I could have returned null and not throw. Should I change it to fallback with null or leave it as it is now?

Reply:
UPDATE: I changed it to fallback to new `getObjQualifiers` and returning the null. This way the deprecated method falls back to single implementation and in case of changes nothing would get forgotten with logic implemented in two places and I left the original behavior of the method unchanged.